Add cypress introduction in fr and en (#53)
lunatech-labs/lunatech-blog: posts/2022-06-06-introduction-cypress-en.adoc

= Introducing Cypress
VincentBrule
v1.0, 2022-06-06
:title: Introducing Cypress
:imagesdir: ../media/2022-06-06-introduction-cypress
:lang: en
:tags: [test, cypress, web, js, ts]

This article is the beginning of a series of three articles dedicated to https://www.cypress.io[Cypress]:

. Introducing Cypress
. Going Further with Cypress
. Comparing Cypress and its Competitors

> The goal of this first article is to make you want to discover and use *Cypress*. 

Reviewer: The goal of this first article is to encourage you to discover and use *Cypress*.

> The goal of this first article is to make you want to discover and use *Cypress*. 

Reviewer: ```suggestion
The goal of this first article is to encourage you to discover and use *Cypress*.
```

To do so, we will present this tool and put it into practice with a simple example.

== I/ What is Cypress used for?

Tests are and always have been an important point in a project. 
There are many types of testing such as unit testing and integration testing. 
Cypress focuses on end-to-end testing (E2E).
The E2E tests allow us to validate the user experience through predefined scenarios. 
These scenarios will describe a group of actions that an average user will perform on our site. 

> This kind of test does not care about the internal implementation of our product, but will adopt the point of view of a user to verify the general behavior of our product.

Reviewer: ```ssuggestion
These tests do not care about the internal implementation of our product, but will adopt the point of view of a user to verify the general behaviour of our product.
```
It's perhaps a little pedantry, but in the previous sentences you refer to test (plural), then suddenly you switch to the singular. No biggy if you leave it, just change the typo "behavior" to "behaviour"

Author: That's a good point, thanks 👍

= Introducing Cypress
VincentBrule
v1.0, 2022-06-06
:title: Introducing Cypress
:imagesdir: ../media/2022-06-06-introduction-cypress
:lang: en
:tags: [test, cypress, web, js, ts]

This article is the beginning of a series of three articles dedicated to https://www.cypress.io[Cypress]:

. Introducing Cypress
. Going Further with Cypress
. Comparing Cypress and its Competitors

The goal of this first article is to encourage you to discover and use *Cypress*.
To do so, we will present this tool and put it into practice with a simple example.

== I/ What is Cypress used for?

> Tests are and always have been an important point in a project. 

Reviewer 2: ```suggestion
Tests are and always have been important in a project. 
```

++++

Let's move on to a practical case study to see how to use Cypress and hopefully make you want to try it too!

== II/ Case study
=== 1. Preparation

For this case study, we will use the Lunatech blog: https://blog.lunatech.com/. 
We will implement the following scenario:

. Access to the blog
. Check that there is at least one article on the home page
. Filter by author X
. Check that all remaining articles are made by author X

The first thing to do is to open our JS/TS project and https://docs.cypress.io/guides/getting-started/installing-cypress[install Cypress].

It's as simple as that:
[source, Shell]
----
yarn add cypress --dev
OR
npm install cypress --save-dev
----

For the example, we will put all our logic in the `cypress` folder and test with the `chrome` browser.
To save you time and not type the full path to the Cypress executable, I suggest you put this script in your `package.json` :
[source, Shell]
----
  "scripts": {
    "cypress:open": "cypress open"
  }
----

[quote, How to organize the code?]

> In this first article about Cypress, we will organize the code very simply, but in the following article we will explore good practices to have a clean and maintainable code over time.

Reviewer 2: ```suggestion
In this first article about Cypress, we will organize the code very simply, but in the following article we will explore good practices to ensure clean and maintainable code over time.
```

You can now run:
[source, shell]
----
npm run cypress:open
----

This command will open a graphical interface to interact with Cypress.
You should see the screen below:

image::first-screen-cypress.png[Home page, width = 700]

On this screen, you can choose *E2E testing* (we will come back to all the subtleties of Cypress in the next article including *Component Testing*).

image::browser-cypress.png[Browser selection, width = 700]

You can choose different browsers. 
In our case, we will use _Chrome_.
Cypress will then create a base project to start your tests.
If all went well, you should have this file structure:

image::fichiers-cypress.png[Files created by Cypress, width = 300]

Now you have everything in place to move on to the practical case. 
Don't close the GUI so you can see your changes in real time!

> === 2. Realization of the practical case

Reviewer 2: ```suggestion
=== 2. Implementation
```
This is the simplest way to express what i _think_ you mean here - if you'd prefer something less succinct then, perhaps:

```suggestion
=== 2. Building out the example
```
